plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/linearization/AssumptionTracker.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.linearization
+
+import org.jetbrains.kotlin.formver.viper.ast.Exp
+
+class AssumptionTracker {
+    val assumptions = mutableListOf<Exp>()
+
+    fun addAssumption(assumption: Exp) {
+        if (assumption !in assumptions) {
```

Comment:
Why not to use a `MutableSet`? If the order of the elements is important you can also use a `LinkedHashSet`

Reply:
Oh, that's a good point, thanks.  (The order is indeed important.)